odoo/addons/base/models/ir_cron.py

```diff
@@ -80,7 +80,8 @@ def default_get(self, fields_list):
     def method_direct_trigger(self):
         self.check_access_rights('write')
         for cron in self:
-            self.with_user(cron.user_id).ir_actions_server_id.run()
+            cron.with_user(cron.user_id).with_context(lastcall=cron.lastcall).ir_actions_server_id.run()
+            cron.lastcall = fields.Datetime.now()
```

Comment:
Isn't it intended that lastcall is only updated by automated runs ? cc @xmo-odoo

Reply:
I don't think so, the goal of lastcall is to provide an easily accessible range of processable records, a manual execution would have processed records up to its own execution so it'd make sense that it would handle and update lastcall correspondingly. Would probably make sense to fix that in 13.0 or even 12.3.

Reply:
@odony what dya think ?

Reply:
Makes sense indeed.

I'm not so fond of the use of a namespace-less `lastcall` context key, but that's not your fault, and it's too late ;-)